### DISCUSSION TITLE
What is the purpose of "standalone" folder?

### DISCUSSION THREAD

**Discussion by janjurca** [TheLartians/Greeter#179] [Q&A]
Hello. 
I am just wondering what is the purpose of standalone folder. Is it for cpp source files? If so why is there a "sources" folder. Why is it split?

Thanks!

**Comment by TheLartians** [comment]
Hey, thanks for the question! This is indeed not well documented. The project defines reusable library `Greeter` in the root directory, consisting of the `source` and `include` directories, that is being used by multiple derived projects. One of them is contained in the `test` directory for unit testing the library and another is inside the `standalone` directory, which showcases a small CLI tool using the library. Each of the projects have their own `CMakeLists.txt` file in their respective directories. This is a bit different than conventional CMake architectures where a single CMake file defines multiple projects which can be enabled / disabled using flags, one big advantage being that it makes the library itself much easier to use for consumers using tools like `CPM.cmake`.
